presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -1272,28 +1321,61 @@ public PrestoS3OutputStream(
 
-            transferManager = TransferManagerBuilder.standard()
-                    .withS3Client(requireNonNull(s3, "s3 is null"))
-                    .withMinimumUploadPartSize(multiPartUploadMinPartSize)
-                    .withMultipartUploadThreshold(multiPartUploadMinFileSize).build();
-
-            requireNonNull(aclType, "aclType is null");
-            requireNonNull(s3StorageClass, "s3StorageClass is null");
-            this.aclType = aclType.getCannedACL();
+            this.s3 = requireNonNull(s3, "s3 is null");
             this.host = requireNonNull(host, "host is null");
             this.key = requireNonNull(key, "key is null");
             this.tempFile = tempFile;
             this.sseEnabled = sseEnabled;
             this.sseType = requireNonNull(sseType, "sseType is null");
             this.sseKmsKeyId = sseKmsKeyId;
-            this.s3StorageClass = s3StorageClass.getS3StorageClass();
+            this.multiPartUploadMinFileSize = multiPartUploadMinFileSize;
+            this.multiPartUploadMinPartSize = multiPartUploadMinPartSize;
+            this.aclType = convertToObjectCannedACL(requireNonNull(aclType, "aclType is null"));
+            this.s3StorageClass = convertToStorageClass(requireNonNull(s3StorageClass, "s3StorageClass is null"));
 
             log.debug("OutputStream for key '%s' using file: %s", key, tempFile);
         }
 
+        private static ObjectCannedACL convertToObjectCannedACL(PrestoS3AclType aclType)
+        {
+            switch (aclType) {
+                case PRIVATE:
+                    return ObjectCannedACL.PRIVATE;
+                case PUBLIC_READ:
+                    return ObjectCannedACL.PUBLIC_READ;
+                case PUBLIC_READ_WRITE:
+                    return ObjectCannedACL.PUBLIC_READ_WRITE;
+                case AUTHENTICATED_READ:
+                    return ObjectCannedACL.AUTHENTICATED_READ;
+                case BUCKET_OWNER_FULL_CONTROL:
+                    return ObjectCannedACL.BUCKET_OWNER_FULL_CONTROL;
+                case BUCKET_OWNER_READ:
+                    return ObjectCannedACL.BUCKET_OWNER_READ;
+                default:
+                    return ObjectCannedACL.PRIVATE;
+            }
+        }
+
+        private static StorageClass convertToStorageClass(PrestoS3StorageClass storageClass)
+        {
+            switch (storageClass) {
+                case STANDARD:
+                    return StorageClass.STANDARD;
+                case REDUCED_REDUNDANCY:
+                    return StorageClass.REDUCED_REDUNDANCY;
+                case GLACIER:
+                    return StorageClass.GLACIER;
+                case DEEP_ARCHIVE:
+                    return StorageClass.DEEP_ARCHIVE;
+                case INTELLIGENT_TIERING:
+                    return StorageClass.INTELLIGENT_TIERING;
+                default:
+                    return StorageClass.STANDARD;
```

Comment:
Based on Presto's documentation we only support `STANDARD` and `INTELLIGENT_TIERING`